upload/src/addons/SV/WarningImprovements/XF/Entity/UserChangeTemp.php

```diff
@@ -45,11 +47,22 @@ public function getResult()
         switch ($this->action_type)
         {
             case 'groups':
-                $result = 'sv_warning_action_added_to_user_groups';
+                $result = 'n_a';
+
+                /** @var \SV\WarningImprovements\XF\Repository\UserChangeTemp $userGroupRepo */
+                $userGroupRepo = $this->repository('XF:UserChangeTemp');
+                $userGroups = $userGroupRepo->getCachedUserGroupsList();
+
+                if (substr($this->action_modifier, 0, 15) === 'warning_action_')
+                {
+                    $userGroupId = intval(substr($this->action_modifier, 15));
```

Comment:
The id on the end isn't the group id but the warning action id.

You need to parse the $this->new_value which is a comma separated list of user group id's

Reply:
Looks like you need to hit xf_user_group_change to get a list of group changes for the action_modifier